src/blas/backends/rocblas/rocblas_level1.cpp

```diff
@@ -0,0 +1,1758 @@
+
+            auto a_ = reinterpret_cast<rocDataType1 *>(a);
+            auto b_ = reinterpret_cast<rocDataType1 *>(b);
+            auto c_ = reinterpret_cast<rocDataType2 *>(c);
+            auto s_ = reinterpret_cast<rocDataType1 *>(s);
+            rocblas_status err;
+            ROCBLAS_ERROR_FUNC(func, err, handle, a_, b_, c_, s_);
+        });
+    });
+    return done;
+}
+
+#define ROTG_LAUNCHER_USM(TYPE1, TYPE2, ROCBLAS_ROUTINE)                                 \
+    cl::sycl::event rotg(cl::sycl::queue &queue, TYPE1 *a, TYPE1 *b, TYPE2 *c, TYPE1 *s, \
+                         const cl::sycl::vector_class<cl::sycl::event> &dependencies) {  \
+        return rotg(ROCBLAS_ROUTINE, queue, a, b, c, s, dependencies);                   \
+    }
+
+ROTG_LAUNCHER_USM(float, float, rocblas_srotg)
+ROTG_LAUNCHER_USM(double, double, rocblas_drotg)
+ROTG_LAUNCHER_USM(std::complex<float>, float, rocblas_crotg)
+ROTG_LAUNCHER_USM(std::complex<double>, double, rocblas_zrotg)
+#undef ROTG_LAUNCHER_USM
+
+template <typename Func, typename T>
+inline cl::sycl::event rotm(Func func, cl::sycl::queue &queue, int64_t n, T *x, int64_t incx, T *y,
+                            int64_t incy, T *param,
+                            const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
+    using rocDataType = typename RocEquivalentType<T>::Type;
+    overflow_check(n, incx, incy);
+    auto done = queue.submit([&](cl::sycl::handler &cgh) {
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        onemkl_rocblas_host_task(cgh, queue, [=](RocblasScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+
+            auto x_ = reinterpret_cast<rocDataType *>(x);
+            auto y_ = reinterpret_cast<rocDataType *>(y);
+            auto param_ = reinterpret_cast<rocDataType *>(param);
+            rocblas_status err;
+            ROCBLAS_ERROR_FUNC(func, err, handle, n, x_, incx, y_, incy, param_);
+        });
+    });
+    return done;
+}
+
+#define ROTM_LAUNCHER_USM(TYPE, ROCBLAS_ROUTINE)                                            \
+    cl::sycl::event rotm(cl::sycl::queue &queue, int64_t n, TYPE *x, int64_t incx, TYPE *y, \
+                         int64_t incy, TYPE *param,                                         \
+                         const cl::sycl::vector_class<cl::sycl::event> &dependencies) {     \
+        return rotm(ROCBLAS_ROUTINE, queue, n, x, incx, y, incy, param, dependencies);      \
+    }
+
+ROTM_LAUNCHER_USM(float, rocblas_srotm)
+ROTM_LAUNCHER_USM(double, rocblas_drotm)
+#undef ROTM_LAUNCHER_USM
+
+template <typename Func, typename T>
+inline cl::sycl::event copy(Func func, cl::sycl::queue &queue, int64_t n, const T *x, int64_t incx,
+                            T *y, int64_t incy,
+                            const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
```

Comment:
Here and other locations:
```suggestion
                            const std::vector<cl::sycl::event> &dependencies) {
```